### ISSUE TITLE
Move Archives and Live LCD News to utilities.

### ISSUE THREAD

**Issue by tanmaydixit** [MobileApplicationsClub/DoJMA#79]
The archives and LCD news are parts of the nav drawer for now.
Move Archives and Live LCD News to utilities fragment as another 2 cards.
link for LCD news is cc.bits-goa.com/enotice/Lcd.php

**Comment by Rushi98** [comment]
@tanmaydixit  `cc.bits-goa.com/enotice/Lcd.php` doesn't work

**Comment by tanmaydixit** [comment]
It works on bits net. Its an intranet site.

**Comment by aryanagarwal15** [comment]
@tanmaydixit We have a lot of archives dating back to 2007. I have data for all in PDF Format for those archive. Instead of archives section pointing towards issue.com link can we have a separate activity where we can show thumbnails of all the archives and an in app PDF reader?

Also it would be better if we just move Live LCD news to links tab. This way you won't have to create extra section for that,

**Comment by Rushi98** [comment]
In app PDF reader isn't possible to implement easily.

**Comment by aryanagarwal15** [comment]
Oh Ok. then we can just let them download the issue in PDF Format and read on in-phone pdf reader.

**Comment by Agrim1234** [comment]
on it

**Comment by aryanagarwal15** [comment]
@Agrim1234 can you please inform on how are you planning to implement archives? I have a lot of PDF dating back to 2007. I want the pdfs to be downloadable. also users should get to know if they have previously downloaded any PDF or not (With a tick right or something). Also all PDFs should automatically detect cover as thumbnail. The recycler view should show The thumbnail and Issue Name. The data needs to be pulled from firebase.

**Comment by Rushi98** [comment]
@aryanagarwal15 Is it possible for you to upload thumbnails for the pdfs as well?

**Comment by aryanagarwal15** [comment]
@Rushi98 sure. I'll make that happen. Please suggest optimum width and height.

**Comment by Rushi98** [comment]
depends on UI but a decent one with 16:9 ish aspect ratio and  would mostly work and refering to [this](https://material.io/tools/devices/) width 1440 px seems maximum on smartphone. So 1440 x 2560 px. @Agrim1234 please confirm.

**Comment by Agrim1234** [comment]
I think 4:3 would be better

**Comment by Agrim1234** [comment]
and most optimum

**Comment by aryanagarwal15** [comment]
The PDFs are in vertical format. So Please suggest Vertical aspect ratio. Also @Agrim1234 please note the archive recycler view should show two issues side by side.

**Comment by Agrim1234** [comment]
I think 1:1 would be best. rest u can decide.
![screenshot-2018-7-7 vertical aspect ratio - google search](https://user-images.githubusercontent.com/31161574/42409639-fb5e9dec-81fa-11e8-8d62-f3c3e1aee64a.png)

**Comment by aryanagarwal15** [comment]
Lets go with 2:3. It looks good.

**Comment by Rushi98** [comment]
@aryanagarwal15 If you have a design in mind tell it now, we will wait

**Comment by Rushi98** [comment]
@Agrim1234 Since @aryanagarwal15 has not responded yet, can you post a mockup thingy of your design (for list) here?

**Comment by aryanagarwal15** [comment]
@Rushi98 I've asked my designers to come up with a design for this. Please give some time. Till then put this on hold.

**Comment by aryanagarwal15** [comment]
Ok so first designs are in. For revamping the UI our designers have come up with this:
<img src="https://user-images.githubusercontent.com/28765515/42466787-8bd56ca6-83cd-11e8-96c1-bf8394b2c9a0.jpeg" width="300">

Here's the link to all svg files used here for icons : https://drive.google.com/open?id=1yXhNmM1bQhi7m8gygl8d64RlvElkumKP
@Rushi98 if you approve this design we can start working towards building this. 
Also please give design credits to "Abhinav Gupta" as he has designed this screen.

More designs to follow.

**Comment by Rushi98** [comment]
looks good 👍 . @tanmaydixit have a look.

**Comment by Agrim1234** [comment]
I think the menus should be another option in utilities just like links and upon clicking on it the menus activity would open and in that activity all these menu would be displayed.

**Comment by aryanagarwal15** [comment]
@Agrim1234 after conducting an internal survey we found out that BITSians majorly use our utilities screen for checking menus. So it made sense to move it to another section. Moreover this looks more organised.

**Comment by aryanagarwal15** [comment]
@Rushi98 More designs are in. Prepared by Abhinav Gupta.

Archives:
<img src="https://user-images.githubusercontent.com/28765515/42572797-fa2d929e-8537-11e8-9cee-2b86ea0f7cf3.jpeg" width="250">

Reorganised Utilities screen:
<img src="https://user-images.githubusercontent.com/28765515/42572830-19eb0706-8538-11e8-8a55-a6dac8616695.jpeg" width="250">

Please approve the designs. Also can you give me a timeframe till when can it be implemented?

**Comment by Agrim1234** [comment]
then it's ok

**Comment by Rushi98** [comment]
@aryanagarwal15 is there something else like "PUBLICATIONS"? If not, it can be removed. Also, the title should be Archive**s**.

**Comment by aryanagarwal15** [comment]
@Rushi98 I agree with both of the mistakes pointed out by you. lets remove "Publications" and change Archive to Archives. I believe Photoshop is used UI design so it is not possible to get exact dimensions. Please use these screenshots for just blueprint of the screens.

**Comment by Agrim1234** [comment]
@aryanagarwal15  give the design for downloading icon(the icon which will appear when the archive is not downloaded) and the progress bar(which will appear when the archive is being downloaded) if u have in mind any otherwise we will go with these
![screenshot-2018-7-14 circular progress bar in android](https://user-images.githubusercontent.com/31161574/42723222-a3959318-8777-11e8-96f8-a46f8049d9bd.png)
![screenshot-2018-7-14 pdf download icon - google search](https://user-images.githubusercontent.com/31161574/42723223-a721849c-8777-11e8-96b4-682051c4bc87.png)

**Comment by aryanagarwal15** [comment]
@Agrim1234 this looks fine. Go ahead with these.

**Comment by aryanagarwal15** [comment]
@Agrim1234 Whats the progress? Please ensure that this is done before 25th. We have to push a launch by then.

**Comment by Agrim1234** [comment]
I have made the archives activity and i will do the utilities UI redesign later.

**Comment by aryanagarwal15** [comment]
Later as in? Also how do you download the pdf in archives? Please share an example.

**Comment by Agrim1234** [comment]
some time later(most probably 10-20 days) as it is not so urgent. Upon clicking on an archive for the first time, archive would be downloaded and stored in the archives folder of external storage directory of dojma. if the user clicks for second time the archive will open in pdf reader app
![screenshot_20180719-215505](https://user-images.githubusercontent.com/31161574/43062539-c9d77284-8e76-11e8-8099-c0a0d838a496.png)
![screenshot_20180719-215510](https://user-images.githubusercontent.com/31161574/43062540-ca5e7e0a-8e76-11e8-90b0-ec7f1b3d70c1.png)
and the  download sign would be like this
![screenshot_20180716-135021](https://user-images.githubusercontent.com/31161574/43062599-03d1ac70-8e77-11e8-80b2-05c4bfa4e757.png)

**Comment by Agrim1234** [comment]
the 2 image shows what will happen after clicking 2 time

**Comment by aryanagarwal15** [comment]
@Agrim1234 How Do I update the archive page? As in where do I put the pdf for archives?

**Comment by Agrim1234** [comment]
See title and image url would be Taken from firebase and the pdf file would be taken from firebase storage as said by rushi

**Comment by aryanagarwal15** [comment]
@Agrim1234 I still cannot access archives. Please share elaborate step by step documentation for adding archive

**Comment by Agrim1234** [comment]
see https://firebase.google.com/docs/storage/android/download-files using the documentation given on this link i created a reference to the archive in firebase storage archives folder and using the method given in download to a local file section i downloaded the archive in the archives folder of the external storage directory related to dojma and after storing it in memory i am using pdf reader app to open it.
And it is working flawlessly in my phone, so everything is right as far as firebase is concerned. I think the problem is memory related.

**Comment by aryanagarwal15** [comment]
@Agrim1234 Sorry man it's not working. I've tested it on multiple devices and emulators. Please fix it asap.

**Comment by Agrim1234** [comment]
I am trying wait for some time.

**Comment by jainvandit99** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/23418345/45980174-e2466580-c06e-11e8-8f6f-5735b3989f10.gif)
Not Fond Of the Design, Took my time to redesign it

**Comment by aryanagarwal15** [comment]
@jainvandit Looks gr8!
